service-base/src/main/java/org/eclipse/hono/service/http/AbstractHttpEndpoint.java

```diff
@@ -92,11 +114,107 @@ protected void extractRequiredJsonPayload(final RoutingContext ctx) {
+        final JsonObject msg = jsonResult.getJsonObject(RequestResponseApiConstants.FIELD_PAYLOAD);
+        if (msg != null) {
+            final String body = msg.encodePrettily();
+            response.putHeader(HttpHeaders.CONTENT_TYPE, HttpUtils.CONTENT_TYPE_JSON_UFT8)
+                    .putHeader(HttpHeaders.CONTENT_LENGTH, String.valueOf(body.length()))
+                    .write(body);
+        }
+    }
+
+    /**
+     * Get a response handler that implements the default behaviour for responding to the HTTP request.
+     *
+     * @param ctx The routing context of the request.
+     * @param action The action of the HTTP request that is being answered.
+     * @param setResponseBodyForStatus A Predicate that defines for which status a response body shall be set. Maybe null.
+     * @param locationFormatter A Function that constructs the String being set as the HTTP location header {@link HttpHeaders#LOCATION}.
+     *                          Must not be null if the HTTP request was to add an object.
+     * @return BiConsumer<Integer, JsonObject> A consumer for the status and the JSON object that implements the default behaviour for responding to the HTTP request.
+     * @throws NullPointerException If ctx is null, if the locationFormatter is null although the request was for adding an object.
+     */
+    protected final BiConsumer<Integer, JsonObject> getDefaultResponseHandler(
+            final RoutingContext ctx,
+            final RequestResponseApiConstants.Action action,
+            final Predicate<Integer> setResponseBodyForStatus,
+            final Function<Void,String> locationFormatter
+            ) {
+
+        final HttpServerResponse response = ctx.response();
+
+        return (status, jsonResult) -> {
+            response.setStatusCode(status);
+            if (status >= 400) {
+                setResponseBody(jsonResult, response);
+            } else if (setResponseBodyForStatus != null) {
+                if (setResponseBodyForStatus.test(status)) {
+                    if (action == RequestResponseApiConstants.Action.ACTION_ADD) {
+                        response.putHeader(
+                                HttpHeaders.LOCATION,
+                                locationFormatter.apply(null));
+                    }
+                    setResponseBody(jsonResult, response);
+                }
+            }
+            response.end();
+        };
+    }
+
+    /**
+     * Sending a request message to a consumer on the event bus for further processing.
+     *
+     * @param ctx The routing context of the request.
+     * @param requestMsg The JSON object to send via the event bus.
+     * @param responseHandler The handler to be invoked for the response received as answer from the event bus.
+     */
+    protected void sendAction(final RoutingContext ctx, final JsonObject requestMsg, final BiConsumer<Integer, JsonObject> responseHandler) {
```

Comment:
is this method supposed to be overridden by subclasses?

Reply:
No, will make it final.